ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/results/BlockResultFactory.java

```diff
@@ -115,6 +118,15 @@ public EngineGetPayloadResultV2 payloadTransactionCompleteV2(
         Quantity.create(blockValue));
   }
 
+  public EngineGetPayloadBodiesResultV1 payloadBodiesCompleteV1(
+      final List<Optional<BlockBody>> blockBodies) {
+    final List<PayloadBody> payloadBodies =
+        blockBodies.stream()
+            .map(maybeBody -> maybeBody.map(blockBody -> new PayloadBody(blockBody)).orElse(null))
```

Comment:
Why are we returning null here?

I would expect that any Optional.empty() blockBodies would be skipped over by the stream, i.e. only the Optional.of() values will be collected into a list...is there a reason to keep the nulls around?

Reply:
Spec says: "3. Client software MUST place null in the response array for unavailable blocks which numbers are lower than a number of the latest known block. " That's why we can't skip the empty blockBodies